launchpad/launchpad.gno

```diff
@@ -0,0 +1,377 @@
+package launchpad
+
+import (
+	"errors"
+	"std"
+	"strconv"
+	"strings"
+	"time"
+
+	"gno.land/p/demo/ufmt"
+	u256 "gno.land/p/gnoswap/uint256"
+	"gno.land/r/gnoswap/v1/common"
+	"gno.land/r/gnoswap/v1/consts"
+	en "gno.land/r/gnoswap/v1/emission"
+	"gno.land/r/gnoswap/v1/gns"
+)
+
+var (
+	projects = make(map[string]Project) // projectId -> project
+
+	// project tier should distrubte project token if deposit ever happend
+	// therefore we need to keep track of project tiers without deposit
+	projectTiersWithoutDeposit = make(map[string]bool) // tierId -> true
+)
+
+var (
+	q96 = u256.Zero()
+)
+
+const (
+	PAD_SEP = "*PAD*"
+)
+
+func init() {
+	q96 = u256.MustFromDecimal(consts.Q96)
+}
+
+// ProjectInput represents the input parameters for creating a new project
+type ProjectInput struct {
+	Name             string
+	TokenPath        string
+	Recipient        std.Address
+	DepositAmount    uint64
+	ConditionsToken  string
+	ConditionsAmount string
```

Comment:
those two strings should be separated with *PAD*
adding comments will be batter

Reply:
https://github.com/gnoswap-labs/gnoswap/pull/451/commits/eb596b597cbe9e8f65f9b07c5802e3767773ddb1